internal/pipe/build/build.go

```diff
@@ -226,5 +226,25 @@ func extFor(target string, flags config.FlagArray) string {
 	if target == "js_wasm" {
 		return ".wasm"
 	}
+	if strings.Contains(target, "linux") {
+		for _, s := range flags {
+			if s == "-buildmode=c-shared" {
+				return ".so"
+			}
+			if s == "-buildmode=c-archive" {
+				return ".a"
+			}
+		}
+	}
+	if strings.Contains(target, "darwin") {
```

Comment:
should we add windows dlls as well?

Reply:
This was implemented before, in PR #1243 - I can reorder the blocks though if you like, maybe in alphabetical order?

Reply:
yes, I think its good to have them all closer together

Reply:
I reordered them now, also in the tests for consistency